testlapack/dlartg.go

```diff
@@ -8,29 +8,81 @@ import (
 	"math"
 	"math/rand"
 	"testing"
+
+	"github.com/gonum/floats"
 )
 
 type Dlartger interface {
 	Dlartg(f, g float64) (cs, sn, r float64)
 }
 
 func DlartgTest(t *testing.T, impl Dlartger) {
+	const (
+		tol   = 1e-14
+		safmn = 2.002083095183101e-146
```

Comment:
Maybe just copy-paste the expressions since they will be simpler in a few minutes.